### ISSUE TITLE
AWS firewall creation fails

### ISSUE THREAD

**Issue by parasj** [skyplane-project/skyplane#495]
`$ skyplane replicate-random gcp:us-central1-a aws:sa-east-1 -n 8 -s $((32)) --chunk-size-mb 4 --debug`

client.log:
```
existing rule for 0.0.0.0/0:22 in skyplane, not adding again
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_iam_role at 0x2a171a820> completed in 0.48s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_iam_role at 0x2a17d20d0> completed in 0.49s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_iam_role at 0x2a1857160> completed in 0.30s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_iam_role at 0x2a1744dc0> completed in 0.30s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_iam_role at 0x2a1777430> completed in 0.31s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_iam_role at 0x2a179fa60> completed in 0.31s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_iam_role at 0x2a1827b80> completed in 0.32s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_iam_role at 0x2a17fc5e0> completed in 0.32s
17:29:15 [DEBUG] [wait_for] Wait for RUNNING status on gcp:us-central1-a:skyplane-gcp-f4112ce5106b40f38d59f16ae55d1b07 fn=<function GCPCloudProvider.provision_instance.<locals>.<lambda> at 0x2ad12d280> completed in 0.74s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_instance_profile at 0x2a8ff9670> completed in 0.10s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_instance_profile at 0x2aa5aeee0> completed in 0.11s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_instance_profile at 0x29cf40a60> completed in 0.10s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_instance_profile at 0x29cf5b9d0> completed in 0.11s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_instance_profile at 0x29cf409d0> completed in 0.10s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_instance_profile at 0x29cf40940> completed in 0.11s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_instance_profile at 0x29cf40af0> completed in 0.11s
17:29:15 [DEBUG] [wait_for] Waiting fn=<function AWSCloudProvider.provision_instance.<locals>.check_instance_profile at 0x29cf40ca0> completed in 0.13s
17:29:15 [DEBUG] [wait_for] Wait for RUNNING status on gcp:us-central1-a:skyplane-gcp-9e809ed8fd414275964644ec8305921d fn=<function GCPCloudProvider.provision_instance.<locals>.<lambda> at 0x2a14eb280> completed in 0.69s
17:29:17 [DEBUG] [wait_for] Wait for RUNNING status on gcp:us-central1-a:skyplane-gcp-83f13a632b2a4605855b1ff4aea50f82 fn=<function GCPCloudProvider.provision_instance.<locals>.<lambda> at 0x2911bb040> completed in 1.32s
17:29:17 [DEBUG] [wait_for] Wait for RUNNING status on gcp:us-central1-a:skyplane-gcp-b7a9b0995be7489e8678a35395c66095 fn=<function GCPCloudProvider.provision_instance.<locals>.<lambda> at 0x29ace4d30> completed in 1.24s
17:29:17 [DEBUG] [wait_for] Wait for RUNNING status on gcp:us-central1-a:skyplane-gcp-b953444d267d4f9b898327171675d166 fn=<function GCPCloudProvider.provision_instance.<locals>.<lambda> at 0x291466d30> completed in 1.19s
17:29:17 [DEBUG] [wait_for] Wait for RUNNING status on gcp:us-central1-a:skyplane-gcp-efb0ee7221774df3bde14a5244627cc6 fn=<function GCPCloudProvider.provision_instance.<locals>.<lambda> at 0x29d2e68b0> completed in 0.95s
17:29:17 [DEBUG] [wait_for] Wait for RUNNING status on gcp:us-central1-a:skyplane-gcp-553ad30c61f7435f9925f1acb093defb fn=<function GCPCloudProvider.provision_instance.<locals>.<lambda> at 0x2a0d2ae50> completed in 0.96s
17:29:17 [DEBUG] [wait_for] Wait for RUNNING status on gcp:us-central1-a:skyplane-gcp-38749073f87648a6874288df284f01dd fn=<function GCPCloudProvider.provision_instance.<locals>.<lambda> at 0x29d82c310> completed in 0.96s
17:29:34 [DEBUG] [wait_for] Waiting for gcp:us-central1-a:skyplane-gcp-9e809ed8fd414275964644ec8305921d to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x2a14eb280> completed in 18.44s
17:29:34 [DEBUG] [wait_for] Waiting for gcp:us-central1-a:skyplane-gcp-efb0ee7221774df3bde14a5244627cc6 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x29d2e68b0> completed in 16.99s
17:29:34 [DEBUG] [wait_for] Waiting for gcp:us-central1-a:skyplane-gcp-b953444d267d4f9b898327171675d166 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x291466d30> completed in 17.43s
17:29:34 [DEBUG] [wait_for] Waiting for gcp:us-central1-a:skyplane-gcp-f4112ce5106b40f38d59f16ae55d1b07 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x2ad12d280> completed in 19.18s
17:29:34 [DEBUG] [wait_for] Waiting for gcp:us-central1-a:skyplane-gcp-83f13a632b2a4605855b1ff4aea50f82 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x2911bb040> completed in 17.91s
17:29:35 [DEBUG] [wait_for] Waiting for gcp:us-central1-a:skyplane-gcp-b7a9b0995be7489e8678a35395c66095 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x29ace4d30> completed in 18.34s
17:29:36 [DEBUG] [wait_for] Waiting for gcp:us-central1-a:skyplane-gcp-553ad30c61f7435f9925f1acb093defb to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x2a0d2ae50> completed in 18.55s
17:29:36 [DEBUG] [wait_for] Waiting for gcp:us-central1-a:skyplane-gcp-38749073f87648a6874288df284f01dd to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x29d82c310> completed in 18.53s
17:29:45 [DEBUG] [wait_for] Waiting for aws:sa-east-1:i-0c19ee68b5f1295e6 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x2948e9af0> completed in 1.28s
17:29:45 [DEBUG] [wait_for] Waiting for aws:sa-east-1:i-058ff7e2031a9ed76 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x2954140d0> completed in 1.22s
17:29:46 [DEBUG] [wait_for] Waiting for aws:sa-east-1:i-0cf4d18a611aaf261 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x29b9ef430> completed in 1.49s
17:29:46 [DEBUG] [wait_for] Waiting for aws:sa-east-1:i-02f5103b073ec2fb5 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x297a07700> completed in 1.52s
17:29:46 [DEBUG] [wait_for] Waiting for aws:sa-east-1:i-0db524df3f90871ba to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x2a17d2040> completed in 1.50s
17:29:46 [DEBUG] [wait_for] Waiting for aws:sa-east-1:i-02b34891f747b1da6 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x29d2114c0> completed in 1.52s
17:29:47 [DEBUG] [wait_for] Waiting for aws:sa-east-1:i-075b79338b8ff23c2 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x29c076f70> completed in 1.69s
17:29:53 [DEBUG] [wait_for] Waiting for aws:sa-east-1:i-000856e22d2643bc4 to be ready fn=<function Server.wait_for_ssh_ready.<locals>.is_up at 0x296e840d0> completed in 1.28s
17:30:01 [DEBUG] [AWS] Adding IPs ['34.136.113.16', '18.228.188.252', '34.71.66.212', '15.228.221.126', '34.68.163.96', '15.228.205.240', '18.230.66.55', '34.170.133.236', '54.233.81.91', '34.170.104.131', '18.230.85.139', '104.197.13.4', '34.67.217.158', '34.72.181.127', '18.231.171.213', '18.229.149.47'] to security group skyplane
17:30:01 [ERROR] [AWS] Error adding IPs ['34.136.113.16', '18.228.188.252', '34.71.66.212', '15.228.221.126', '34.68.163.96', '15.228.205.240', '18.230.66.55', '34.170.133.236', '54.233.81.91', '34.170.104.131', '18.230.85.139', '104.197.13.4', '34.67.217.158', '34.72.181.127', '18.231.171.213', '18.229.149.47'] to security group skyplane: An error occurred (InvalidPermission.Duplicate) when calling the AuthorizeSecurityGroupIngress operation: the specified rule "peer: 34.67.217.158/32, ALL, ALLOW" already exist
```

**Comment by parasj** [comment]
Mitigations:
* Ideally, this should error out and raise an exception to the user rather than silently erroring out
* Debug why firewall isn't adding rules

**Comment by parasj** [comment]
Fixed by #516
